File: 백준/Silver/9196. 정수 직사각형/정수 직사각형.java
import java.io.*;
import java.util.*;

public class Main {
    private static String findRect(int h, int w) {
        int curDiagnoal = h * h + w * w;
        int minDiagonal = 151 * 151 * 2;
        int resultH = 0;
        int resultW = 0;
        for (int nh = 1; nh <= 150; nh++) {
            for (int nw = nh + 1; nw <= 150; nw++) {
                int newDiagonal = nh * nh + nw * nw;
                if (newDiagonal > curDiagnoal || (newDiagonal == curDiagnoal && nh > h)) {
                    if (newDiagonal < minDiagonal || (newDiagonal == minDiagonal && nh < resultH)) {
                        minDiagonal = newDiagonal;
                        resultH = nh;
                        resultW = nw;
                    }
                }
            }
        }
        String output = resultH + " " + resultW;
        return output;
    }

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int h = sc.nextInt();
            int w = sc.nextInt();
            if (h == 0 && w == 0)
                break;
            System.out.println(findRect(h, w));
        }
        sc.close();
    }
}

File: 백준/Silver/9196. 정수 직사각형/정수 직사각형.py
def findRect(h, w):
    curDiagonal = h**2 + w**2
    minDiagonal = 151**2 * 2
    resultH = 0
    resultW = 0

    for nh in range(1, 151):
        for nw in range(nh + 1, 151):
            newDiagonal = nh**2 + nw**2
            if (newDiagonal > curDiagonal) or (newDiagonal == curDiagonal and nh > h):
                if (newDiagonal < minDiagonal) or (newDiagonal == minDiagonal and nh < resultH):
                    minDiagonal = newDiagonal
                    resultH = nh
                    resultW = nw
                
    return resultH, resultW

while True:
    h, w = map(int, input().split())
    if h == 0 and w == 0:
        break
    print(*findRect(h, w))